### PR TITLE
Add #[must_use] to all functions 'fn(float) -> float'

### PR DESCRIPTION
**Pull request by BatmanAoD**
These are pure functions.

```rust
impl f32/f64 {
    fn floor(self) -> Self;
    fn ceil(self) -> Self;
    fn round(self) -> Self;
    fn trunc(self) -> Self;
    fn fract(self) -> Self;
    fn abs(self) -> Self;
    fn signum(self) -> Self;
    fn mul_add(self, a: Self, b: Self) -> Self;
    fn div_euclid(self, rhs: Self) -> Self;
    fn rem_euclid(self, rhs: Self) -> Self;
    fn powi(self, n: i32) -> Self;
    fn powf(self, n: Self) -> Self;
    fn sqrt(self) -> Self;
    fn exp(self) -> Self;
    fn exp2(self) -> Self;
    fn ln(self) -> Self;
    fn log(self, base: Self) -> Self;
    fn log2(self) -> Self;
    fn log10(self) -> Self;
    fn abs_sub(self, other: Self) -> Self;
    fn cbrt(self) -> Self;
    fn hypot(self, other: Self) -> Self;
    fn sin(self) -> Self;
    fn cos(self) -> Self;
    fn tan(self) -> Self;
    fn asin(self) -> Self;
    fn acos(self) -> Self;
    fn atan(self) -> Self;
    fn atan2(self, other: Self) -> Self;
    fn exp_m1(self) -> Self;
    fn ln_1p(self) -> Self;
    fn sinh(self) -> Self;
    fn cosh(self) -> Self;
    fn tanh(self) -> Self;
    fn asinh(self) -> Self;
    fn acosh(self) -> Self;
    fn atanh(self) -> Self;
    fn clamp(self, min: Self, max: Self) -> Self;
}
```

Part of #48926

### CODE REVIEW

**Comment by rkruppe** [comment]
My impression (confirmed by re-reading the linked issue and others cross-referenced there) was that we don't currently have a precedent for applying `must_use` sweepingly to entire classses of functions, and instead only add it to individual functions when there's concrete motivation for that specific functions being `must_use` (i.e., real world bugs involving that function's result being ignored). This PR deviates from that, which... at least requires broader decision-making than just merging it.

**Comment by BatmanAoD** [comment]
@rkruppe I personally misused `.round()`, but I don't think there's anything special about `round` that makes it more likely to be misused. So while I agree that the last comment in that thread recommends a conservative metric for annotatinmg on a per-function basis only, I think that annotating only `round` would be a code smell at best.

**Comment by killercup** [comment]
I don't have any opinion on whether this should land, but if it does, I'd like there to be a message that makes the error a bit more expressive (something along the lines of `#[must_use = "method returns a new number and does not mutate the caller"]`).

**Comment by BatmanAoD** [comment]
@killercup That sounds good! How about "...mutate the original value" rather than "caller"?

**Comment by BatmanAoD** [comment]
Interesting; it looks like `clamp` is being used without using the return
value.

File: src/libstd/f32.rs [hunk 1]
```diff
@@ -1015,7 +1015,8 @@ impl f32 {
     /// assert!((2.0f32).clamp(-2.0, 1.0) == 1.0);
     /// assert!((std::f32::NAN).clamp(-2.0, 1.0).is_nan());
     /// ```
-    #[must_use = "method returns a new number and does not mutate the original value"]
+    // The tests below invoke `clamp` without a return value in order to cause a `panic`.
```
Comment (fbstj) [inline src/libstd/f32.rs]: are these worthy of `FIXME`?

**Comment by BatmanAoD** [comment]
I think we should either fix the tests now or intentionally leave these
functions un-annotated (in case anyone is using `clamp` for its `panic`
behavior, which...is generally probably not a good practice).

**Comment by BatmanAoD** [comment]
I've gone ahead and restored the annotation for `clamp` and fixed the tests.

**Comment by JohnCSimon** [comment]
Ping from triage
@joshtriplett @fbstj can you please re-review this PR?

CC @BatmanAoD 

Thanks!

**Comment by joshtriplett** [comment]
I'm a little confused why rfcbot didn't do its thing.

Let's try this instead:

@rfcbot poll Merge this?

**Comment by Centril** [comment]
> I'm a little confused why rfcbot didn't do its thing.

The team label was T-libs and you're not on that team. :)

Ping @Amanieu @Kimundi @SimonSapin @sfackler @withoutboats

**Comment by SimonSapin** [comment]
r+ on the diff assuming consensus

**Comment by Dylan-DPC** [comment]
@Amanieu @Kimundi waiting for your consensus on [the above](https://github.com/rust-lang/rust/pull/63871#issuecomment-533402410)

**Comment by JohnTitor** [comment]
Ping from triage: @Kimundi waiting for your consensus

**Comment by Dylan-DPC** [comment]
@SimonSapin this has passed consensus

**Comment by Dylan-DPC** [comment]
Thanks @withoutboats

**Comment by JohnCSimon** [comment]
Ping from triage
@BatmanAoD can you please fix the test failures?
cc: @SimonSapin 
Thanks

**Comment by JohnCSimon** [comment]
Ping again from triage
@BatmanAoD can you please fix the test failures?
cc: @SimonSapin @withoutboats 
Thanks

**Comment by BatmanAoD** [comment]
As far as I can tell, this is a spurious failure:

```
2019-10-27T23:48:07.1759600Z Compiling compiler_builtins v0.1.18
2019-10-27T23:48:07.1798193Z error: failed to remove file `D:\a\1\s\build\i686-pc-windows-gnu\stage1-std\i686-pc-windows-gnu\release\deps\libcompiler_builtins-4f0583857a7f6e60.rlib`
2019-10-27T23:48:07.1799014Z
2019-10-27T23:48:07.1799241Z Caused by:
2019-10-27T23:48:07.1799814Z Access is denied. (os error 5)
```

How do I request a retry?

**Pull request merged**